packages/sanity/src/core/config/configPropertyReducers.ts

```diff
@@ -310,6 +310,27 @@ export const documentCommentsEnabledReducer = (opts: {
   return result
 }
 
+export const onStudioErrorResolver = (opts: {
```

Comment:
Given the way this is used I think this function could handle reporting the error to any of the plugins that defines an errorResolver by iterating in the flatten list and triggering each of the functions.

```ts
export const onStudioErrorResolver = (opts: {
  config: PluginOptions
  context: {error: Error; errorInfo: ErrorInfo}
}) => {
  const {config, context} = opts
  const flattenedConfig = flattenConfig(config, [])
  flattenedConfig.forEach(({config: pluginConfig}) => {
    // There is no concept of 'previous value' in this API. We only care about the final value.
    // That is, if a plugin returns true, but the next plugin returns false, the result will be false.
    // The last plugin 'wins'.
    const resolver = pluginConfig.onStudioError

    if (typeof resolver === 'function') return resolver(context.error, context.errorInfo)
    if (!resolver) return undefined

    throw new Error(
      `Expected \`document.onStudioError\` to be a a function, but received ${getPrintableType(
        resolver,
      )}`,
    )
  })
}

```

Reply:
Another great suggestion which I will be adding :)